src/cascadia/TerminalSettingsEditor/SettingContainer.cpp

```diff
@@ -188,9 +189,19 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
         else if (originTag == Model::OriginTag::Generated)
         {
-            // from a dynamic profile generator
-            // TODO #1690: add special handling for proto-extensions here
-            return {};
+            const auto profileSource{ profile.Source() };
+            if (profileSource == WslGeneratorNamespace ||
+                profileSource == AzureGeneratorNamespace ||
+                profileSource == PowershellCoreGeneratorNamespace)
+            {
+                // from a dynamic profile generator
+                return {};
+            }
+            else
+            {
+                // proto-extensions
+                return hstring{ fmt::format(std::wstring_view(RS_(L"SettingContainer_OverrideMessageProtoExtension")), profileSource) };
+            }

```

Comment:
no.

Reply:
I flatly reject a dependency from this UI element to an implementation detail in the settings model. You're going to have to find a better way to do this.

Maybe you need a "Fragment" origin.

Reply:
ok... can I get a bit more feedback here?

Reply:
Maybe you need to tag things that come from fragments as coming from fragments rather than coming from "Generated" to give them special behavior?

Reply:
The name is not going to be particularly pretty if it comes from a package. It will say, "value from: CanonicalGroupLtd.UbuntuOnWindows_7yabmiilazx1"

Reply:
_ewwww_. Understandable. Is there a way to detect that it came from a package or if it's going to have an ugly name? Maybe some way to polish the ugly name into a cleaner one?

Reply:
I'm ok with doing this for now, then revisiting it after some user feedback. We're going to have to reevaluate all of this work anyways :(